apps/dashr-shiny-diamond-explorer/app.R

```diff
@@ -0,0 +1,191 @@
+    '2000' = list('label' = '2,000')
+  ),
+  value = 1000
+)
+
+pageTitle <- htmlH2('Diamonds Explorer')
+
+plotlyLogo <- htmlA(
+  list(htmlImg(id = 'banner-image', src = 'assets/image.png')),
+  className = 'logo',
+  href = 'https://dashr.plot.ly')
+
+xDropDown <- dccDropdown(
+  id = 'x-dropdown',
+  options = DropDownMenuOptions,
+  value = 'carat',
+  clearable = FALSE
+)
+
+yDropDown <- dccDropdown(
+  id = 'y-dropdown',
+  options = DropDownMenuOptions,
+  value = 'price',
+  clearable = FALSE
+)
+
+colorDropDown <- dccDropdown(
+  id = 'color-dropdown',
+  options = DropDownMenuOptions,
+  value = 'clarity',
+  clearable = FALSE
+)
+
+facetRowDropDown <- dccDropdown(
+  id = 'facet-row-dropdown',
+  options = DropDownMenuOptions,
+  value = 'clarity',
+  clearable = FALSE
+)
+
+facetColumnDropDown <- dccDropdown(
+  id = 'facet-column-dropdown',
+  options = DropDownMenuOptions,
+  value = '.',
+  clearable = FALSE
+)
+
+
+# App Start ------------------------------
+
+# Initiate application
+app = Dash$new()
+
+
+# Create Layout ------------------------------
+
+app$layout(
+  htmlDiv(list(
+    plotlyLogo,
+    pageTitle
+    )),
+  htmlDiv(
+    list(
+      htmlB('Sample Size'),
+      htmlDiv(SampleSlider, style = list('marginBottom' = 40)),
+      htmlB('X'),
+      xDropDown,
+      htmlBr(),
+      htmlB('Y'),
+      yDropDown,
+      htmlBr(),
+      htmlB('Color'),
+      colorDropDown,
+      htmlBr(),
+      htmlB('FacetRow'),
+      facetRowDropDown,
+      htmlBr(),
+      htmlB('FacetColumn'),
+      facetColumnDropDown,
+      htmlBr(),
+      htmlB('Height of plot (in pixels)'),
+      heightSlider
+    ),
+    className = 'three columns'
+  ),
+  htmlDiv(list(dccGraph(id = 'facetgrid')),
+          className = 'eight columns')
+)
+
+
+# Callbacks Start ------------------------------
+
+app$callback(output = list(id = 'facetgrid', property = 'figure'),
+             params = list(
+               input(id = 'sample-slider', property = 'value'),
+               input(id = 'height-slider', property = 'value'),
+               input(id = 'x-dropdown', property = 'value'),
+               input(id = 'y-dropdown', property = 'value'),
+               input(id = 'color-dropdown', property = 'value'),
+               input(id = 'facet-row-dropdown', property = 'value'),
+               input(id = 'facet-column-dropdown', property = 'value')
+             ),
+             
+  function(sampleSize, plotHeight, x, y, color, facet_row,facet_col) {
+
+    #Adding Reactive data info, dataset is built in diamonds data
+    dataset <- diamonds[sample(nrow(diamonds), sampleSize),]
+
+    #FacetGrid ggplot2 object
+    p <- ggplot(dataset, aes_string(x = x, y = y, color = color)) + geom_point()
+
+    # Add it if least one facet column/row is specified
+    facets <- paste(facet_row, '~', facet_col)
+
+
+    if (facets != '. ~ .') { p <- p + facet_grid(facets)
+      p1 <- ggplotly(p) %>%
+      layout(height = plotHeight,
```

Comment:
Fixed `height` argument to specify in the `ggplotly` function, specifying it in `layout()` is deprecated.